hw6.rb

```diff
@@ -0,0 +1,178 @@
+ arr.delete(11)
+arr<<3
+print arr
+
+puts
+# problem 6
+# Get rid of duplicates without specifically removing any one value.
+print arr.uniq
+
+
+#  problem 7
+# Tell what's the major difference between an Array and a Hash?
+#
+# Arrays are ordered
+# Array index starts at 0
+#
+# Hash are unordered
+# Hash is a collection of key-value pairs
+# Hash index can be any object type
+# Hash enumerate their values in the order that corresponding keys were inserted
+puts
+
+#  problem 8
+# Suppose you have a hash h=0 a:1, b:2, c:3, d:4 }
+h = { a:1, b:2, c:3, d:4}
+puts
+
+
+# problem 9
+# Can hash values be arraus? Yes hash = {}; hash = { "1":[1,2,3,], "2":["a", "b", "c"], "c":["ruby",11,22,"rails"]}
+# can you have an array of hashes? (give examples) Yes
+#
+h={"firstname"=>"Nicolae", "lastname"=>"Spinu", "age"=>"30", "city"=>"Moldova"}
+h2={"firstname"=>"Olesea", "lastname"=>"Spinu", "age"=>"25", "city"=>"Chisinau"}
+array =[]
+array << h
+array << h2
+print array
+puts
+
+
+#   problem 10
+# Given the following data structures.
+# Write a program that copies the information from the array into the empty hash that applies to the correct person.
+#
+contact_data = [["Joe@email.com"], ["123 Main st.", "555-123-4567"],["sally@email.com","404 Not Foud Dr.","123-234-3454"]]
+contacts = {"Joe Smith" => {}, "Sally Johnson"=>{}}
+
+element=contact_data[0]
+contacts["Joe Smith"]={"email"=>element[0],"address"=>element[1], "phone"=>element[2]}
+element=contact_data[1]
+contacts["Sally Johnson"]={"email"=> element[0], "address"=> element[1],"phone"=> element[2]}
+puts contacts
+puts
+
+
+# problem 11
+# Using the hash you created from the previous exercise,
+# demonstrate how you would access Jos's email and Sally's phone nember?
+puts contacts ["Joe Smith"]["email"]
+puts contacts ["Sally Johnson"]["phone"]
+puts
+
+# problem 12
+# Use Ruby's Array delete if and String method start with?
+# to delete all of the words that begin with an "s" in the following array.
+# arr=['snow', 'winter', 'ice','slippery','salted roads','white trees']
+arr = ['snow', 'winter', 'ice', 'slippery','salted roads','white trees']
+arr.delete_if { |word| word.start_with?("s") }
+print arr
+puts
+
+# Then recreate the arr and get rid of all of the words that with "s" or starts with "w".
+arr = ['snow', 'winter', 'ice', 'slippery', 'salted roads', 'white trees']
+arr.delete_if { |word|word.start_with?("s","w") }
+print arr
+puts
+
+# problem 13
+# Take the following array:
+# a = ['white snow','winter wonderland','melting ice','slippery sidewalk','salted roads','wite trees']
+# and turn it into a new araay that consists of strings containing one word.
+# (ex.["wite snow", etc...]->["wite","snow", etc...])
+# Look into using Array's map and flatten methods,as well as String's method.
+a = ['white snow', 'winter wonderland','melting ice', 'slippery sidewalk', 'salted roads', 'wite trees']
+arr = []
+# arr+=a.each{|el| el.split(" ")}   Please Explain Why not working ????
```

Comment:
a.each { |el| el.split(" ") } returns a unchanged, you can try arr = a.map { |el| el.split(" ") }